src/uct/ib/base/ib_device.h

```diff
@@ -61,7 +61,8 @@ enum {
     UCT_IB_DEVICE_FLAG_DC_V2    = UCS_BIT(7),   /* Device supports DC ver 2 */
     UCT_IB_DEVICE_FLAG_AV       = UCS_BIT(8),   /* Device supports compact AV */
     UCT_IB_DEVICE_FLAG_DC       = UCT_IB_DEVICE_FLAG_DC_V1 |
-                                  UCT_IB_DEVICE_FLAG_DC_V2 /* Device supports DC */
+                                  UCT_IB_DEVICE_FLAG_DC_V2, /* Device supports DC */
+    UCT_IB_DEVICE_FLAG_KSM      = UCS_BIT(9)    /* Device supports KSM */
```

Comment:
maybe it's time to extend uct_ib_md_t to uct_ib_mlx5_md_t, and add this flag there?

Reply:
Need to check for KSM whether it's mlx5 or not.
Look here  https://github.com/openucx/ucx/pull/3095/files#diff-734c9882bf0ceede1a096f651a5f2204R93

Reply:
you mean that would need to add mlx5 check to the gtest? this is not a problem.